File: iosApp/iosApp/Services/AnalyticImpl.swift
import Foundation
import shared

class AnalyticImpl: Analytic {
    private let logger: Logger
    
    init(logger: Logger) {
        self.logger = logger
    }
    
    func logEvent(event: String) {
        logger.log(text: "Event \"\(event)\" sent to analytic by iOS implementation")
    }
}

File: androidApp/src/main/java/com/example/kmpdiexample/android/services/AnalyticImpl.kt
package com.example.kmpdiexample.android.services

import com.example.kmpdiexample.services.Analytic
import com.example.kmpdiexample.services.Logger

class AnalyticImpl(private val logger: Logger): Analytic {
    override fun logEvent(event: String) {
        logger.log("Event \"$event\" sent to analytic by Android implementation")
    }
}